### ISSUE TITLE
Real space is incorrect

### ISSUE THREAD

**Issue by mscroggs** [FEniCS/dolfinx#1208]
In old dolfin, a Real space has one dof in total. In dolfinx, this space appears to have one DOF per cell

```
import numpy as np
from mpi4py import MPI
import dolfinx
from dolfinx.cpp.mesh import CellType
from dolfinx import (RectangleMesh, FunctionSpace)
from ufl import FiniteElement 

mesh = RectangleMesh(
    MPI.COMM_WORLD,
    [np.array([0, -0.5, 0]), np.array([10, 0.5, 0])], [1, 2],
    CellType.quadrilateral, dolfinx.cpp.mesh.GhostMode.none)

R = FiniteElement("Real", mesh.ufl_cell(), 0)
space = FunctionSpace(mesh, R)

assert space.dim == 1
```

This likely is related to [this issue](https://fenicsproject.discourse.group/t/problem-interpolating-mixed-function-dolfinx/4142/6) on discourse: interpolating into a MixedElement including a Real causes a SegFault:

```
import numpy as np
from mpi4py import MPI
from petsc4py import PETSc
import dolfinx
from dolfinx.cpp.mesh import CellType
from dolfinx import (RectangleMesh, Function, FunctionSpace)
from ufl import (FiniteElement, MixedElement, split)

mesh = RectangleMesh(
    MPI.COMM_WORLD,
    [np.array([0, -0.5, 0]), np.array([10, 0.5, 0])], [5, 5],
    CellType.quadrilateral, dolfinx.cpp.mesh.GhostMode.none)

N = FiniteElement("CG", mesh.ufl_cell(), 1)
R = FiniteElement("Real", mesh.ufl_cell(), 0)
W = FunctionSpace(mesh, MixedElement([N, R]))

w = Function(W)

def w_init(x):
    values = np.zeros((2, x.shape[1]))
    values[0] = 0.0
    values[1] = 0.0
    return values

w.interpolate(w_init)
```

**Comment by chrisrichardson** [comment]
I think @michalhabera may have some idea about this. There has been an issue with "Real" spaces for some time. They are not very scalable, resulting in dense rows in matrices, and difficult in parallel.

**Comment by michalhabera** [comment]
Duplicate of https://github.com/FEniCS/dolfinx/issues/225

"Global" spaces need some serious attention and discussion. In the old code, these spaces stored "number of global dofs" as extra information in the generated code. This made dofmap construction uglier. Also, the "Real" space is just a hack to try to fit a non locally supported basis into the scope of FIAT, which is all about local support.

I think we should not abuse and pollute the local FE framework with global spaces. We should think about how to incorporate them purely in dolfinx level.

**Comment by mscroggs** [comment]
Ok. closing this as its a duplicate
